### PR TITLE
NIFI-3607 Allow multi files mode with fixed names

### PR DESCRIPTION
**Pull request by pvillard31**
Thank you for submitting a contribution to Apache NiFi.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [X] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [X] Does your PR title start with NIFI-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [X] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [X] Is your initial contribution a single, squashed commit?

### For code changes:
- [X] Have you ensured that the full suite of tests is executed via mvn -Pcontrib-check clean install at the root nifi folder?
- [X] Have you written or updated unit tests to verify your changes?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

=============================================

I slightly updated the test framework to allow a run schedule duration setting. It allows me to schedule multiple calls to the ``onTrigger`` method with a given duration between each call so that I can check what happens in case of events occurring while the processor is running (without calling ``onScheduled``). With this new parameter, I added a unit test where a new file is created while the processor is running to check that the processor picks up the new file as expected.

### CODE REVIEW

**Comment by trixpan** [comment]
will review

File: nifi-mock/src/main/java/org/apache/nifi/util/StandardProcessorTestRunner.java [hunk 1]
```diff
@@ -36,9 +36,9 @@
 import java.util.Objects;
 import java.util.Set;
 import java.util.concurrent.Callable;
-import java.util.concurrent.ExecutorService;
 import java.util.concurrent.Executors;
 import java.util.concurrent.Future;
+import java.util.concurrent.ScheduledExecutorService;
```
Comment (trixpan) [inline nifi-mock/src/main/java/org/apache/nifi/util/StandardProcessorTestRunner.java]: wasn't familiar with this. very handy!

**Comment by trixpan** [comment]
Shouldn't `static final AllowableValue MODE_MULTIFILE` ... be updated to reflect new behavior?

**Comment by pvillard31** [comment]
Sorry for the late answer @trixpan, just coming back from holidays. You're right, I should update the description, will do it later today. Thanks!

**Comment by pvillard31** [comment]
Just pushed a commit to address your comment let me know if you want me to squash the PR.

**Comment by trixpan** [comment]
@pvillard31 sorry for this but it seems PR is now in conflict with master. Would you mind rebasing?
Cheers

**Comment by pvillard31** [comment]
No problem at all @trixpan, I rebased against master.

**Comment by trixpan** [comment]
@pvillard31 thanks for the rebase. LGTM. Merged

**Pull request closed without merging**